plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -4464,6 +4464,61 @@ public void testCreateExternalTableWithNullFormat()
                 ImmutableList.of("null_format = 'NULL_VALUE'"));
     }
 
+    @Test
+    public void testCreateExternalTableOnNonExistingPath()
+            throws Exception
+    {
+        java.nio.file.Path tempDir = createTempDirectory(null);
+        // delete dir, trino should recreate it
+        deleteRecursively(tempDir, ALLOW_INSECURE);
+        String tableName = "test_create_external_non_exists_" + randomNameSuffix();
+
+        @Language("SQL") String createTableSql = format("" +
+                        "CREATE TABLE %s.%s.%s (\n" +
+                        "   col1 varchar,\n" +
+                        "   col2 varchar\n" +
+                        ")\n" +
+                        "WITH (\n" +
+                        "   external_location = '%s',\n" +
+                        "   format = 'TEXTFILE'\n" +
+                        ")",
+                getSession().getCatalog().get(),
+                getSession().getSchema().get(),
+                tableName,
+                tempDir.toUri().toASCIIString());
+
+        assertUpdate(createTableSql);
+        MaterializedResult actual = computeActual(format("SHOW CREATE TABLE %s", tableName));
+        assertEquals(actual.getOnlyValue(), createTableSql);
+        assertUpdate(format("DROP TABLE %s", tableName));
+        deleteRecursively(tempDir, ALLOW_INSECURE);
+    }
+
+    @Test
+    public void testCreateExternalTableOnExistingPathToFile()
+            throws Exception
+    {
+        File tempFile = File.createTempFile("temp", ".tm");
+        tempFile.deleteOnExit();
+        String tableName = "test_create_external_on_file_" + randomNameSuffix();
+
+        @Language("SQL") String createTableSql = format("" +
+                        "CREATE TABLE %s.%s.%s (\n" +
+                        "   col1 varchar,\n" +
+                        "   col2 varchar\n" +
+                        ")\n" +
+                        "WITH (\n" +
+                        "   external_location = '%s',\n" +
+                        "   format = 'TEXTFILE'\n" +
+                        ")",
+                getSession().getCatalog().get(),
+                getSession().getSchema().get(),
+                tableName,
+                tempFile.toPath().toUri().toASCIIString());
+
+        assertQueryFails(createTableSql, "External location is not a valid file system URI.*");
```

Comment:
Here the error message should be meaningful right ? Like `Unable to create, External location must be a directory`

Reply:
refactored